cpp/src/gandiva/function_registry_external.cc

```diff
@@ -0,0 +1,301 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <arrow/type.h>
+#include <gandiva/function_registry_external.h>
+#include <rapidjson/document.h>
+#include <filesystem>
+#include <fstream>
+#include <vector>
+
+namespace gandiva {
+namespace rj = rapidjson;
+
+class JsonRegistryParser {
```

Comment:
A JSON registry parser is implemented to parse the function metadata from the JSON content.